homeassistant/components/prusalink/__init__.py

```diff
@@ -27,16 +35,69 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    """Set up PrusaLink from a config entry."""
+async def _migrate_to_version_2(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> PrusaLink | None:
+    """Migrate to Version 2."""
+    _LOGGER.debug("Migrating entry to version 2")
+
+    data = dict(entry.data)
+    # "maker" is currently hardcoded in the firmware
+    # https://github.com/prusa3d/Prusa-Firmware-Buddy/blob/bfb0ffc745ee6546e7efdba618d0e7c0f4c909cd/lib/WUI/wui_api.h#L19
+    data[CONF_USERNAME] = "maker"
+    data[CONF_PASSWORD] = entry.data[CONF_API_KEY]
+    data.pop(CONF_API_KEY)
+
     api = PrusaLink(
         async_get_clientsession(hass),
-        entry.data["host"],
-        entry.data["api_key"],
+        data[CONF_HOST],
+        data[CONF_USERNAME],
+        data[CONF_PASSWORD],
     )
+    try:
+        await api.get_info()
+    except InvalidAuth:
+        # We don't know for sure if the firmware is actually outdated
+        # If we hit an InvalidAuth error after the user configured this integration
+        # (where we already checked that credentials are correct)
+        # then its most likely an issue with unsupported firmware.
```

Comment:
I would rewrite the comment that we assume that the firmware is outdated as we never successfully connected using the new api

Reply:
I've reworded it. I hope it fits better now.
Thanks!